mdnssd-min.c

```diff
@@ -1015,11 +1017,33 @@ static mDNSservice_t *build_update(struct context_s *context, bool build) {
 		s->status = MDNS_CURRENT;
 	}
 
-	// now we can remove the service
-	if (expired) {
+	if (ptr_expired && srv_expired && txt_expired) {
+		// all RRs for service are expired.
+		// now we can remove the service
 		remove_item((item_t*) s, (item_t**) &context->slist);
 		free_s(s);
 	}
+        else {
+          // Mark expired RRs invalid
+          if (ptr_expired && s->eol[0]) {
+            s->eol[0] = 0;
+            s->status = MDNS_EXPIRED;
+          }
+          if (srv_expired && s->eol[1]) {
+            s->eol[1] = 0;
+            NFREE(s->hostname);
+            s->port = 0;
+            s->hostname = NULL;
```

Comment:
NFREE macro set the freed pointer to NULL

Reply:
Are you sure?  The macro definition send to be [here](https://github.com/philippe44/mDNS-SD/blob/v2/mdnssd-min.h#L29)

Reply:
ahah :-) obviously not. It's a macro I'm using in different applications but obviously in that one I did not add the set to NULL